### DISCUSSION TITLE
Page Break - Margin

### DISCUSSION THREAD

**Discussion by georak** [spatie/laravel-pdf#124] [General]
Hello all,

I have one div with dynamic content which may need more than 1 page to be displayed. Is there any way to add margin top on the second page **only** (without using ->margins())

<img width="1064" alt="image" src="https://github.com/spatie/laravel-pdf/assets/3660446/d128dc17-514b-4141-85ce-d8638b353e63">

**Comment by ArielMejiaDev** [comment]
Hi @georak I think that in blade files you can use `$loop->index` variable to identify the paragraph and apply the page break directive, I suggest to first format the content using a view presenter and then pass it to a blade view, it could make it more easy to handle.

eg:

```php
class YourDocumentViewPresenter
{
    public Collection $pages;

    public function __construct($text)
    {
        $linesArray = explode("\n", $text);

        $linesCollection = collect($linesArray);
        
        $this->pages = $linesCollection->chunk(4);
    }
}
```

In your controller you should be able to use the view presenter like this:

```php
$text = "This is the first line.\nThis is the second line.\nAnd here goes the third line.";

$content = new YourDocumentViewPresenter($text);

return pdf('your-pdf', $content->pages);
``` 

Then in your blade file you should be able to do something like:

```php
@foreach($pages as $page)
    @if($loop->index % 4 == 0)
        @pageBreak
    @endif 
    {{ $page }}
@endforeach
```

With this code you are using a view presenter to transform a text into a collection and then chunk it by groups of 4 (to set a paragraphs that looks like a safe amount of text per paragraph).

Then it loops in a blade file and as the paragraphs used are 4 per page (a safe number of paragraphs per page) you can print the directive to page break if the loop index residual of 4 is equal to zero.

## Take in mind

You can go a little bit further and check the amount of strings per line or per paragraph in the view presenter, I just set a good example but you can set it as granular as you need.

Hope it helps
